### DISCUSSION TITLE
Edit weapons and damage while in map edit mode?

### DISCUSSION THREAD

**Discussion by ShiroWork-s** [redeclipse/base#1262] [Ideas and Feature Requests]
Is it possible to do that? And if not, maybe lets work on adding it! :D

**Comment by qreeves** [comment]
Could you please describe your idea/feature request in more detail?

**Reply by ShiroWork-s** [reply]
Definitely,
When I go into map edit for any map, I can see 
Map; Environment; Multiplayer; and other stuff, but no way to edit any items.
But even when i select items like a grenade or something, there's no way to edit the item itself. I know it may be a long term project to add editing for items like "Shotgun spread" or "SMG damage" or "Grenade radius", but i think it would be a good thing to add for the game.

**Reply by qreeves** [reply]
Ah, I see. We are currently working on an overhaul of the editing interface, and have plans for something similar to what you've described as part of our sandbox mode extension, but the latter may take a year or two before being ready.

Currently, weapon mechanics are controlled by server side game variables and are not controllable by the map. We will be working on providing a menu for changing these variables, but you may be interested in messing around with them in the interim. If you type `/` followed by the name of the weapon, then press TAB, you can cycle through a list of variables for that weapon (eg. `/shotgunspread1`).

I hope this helps you have some fun playing around in the meantime. If you have any further questions please feel free to ask, and you're invited to join us on our Discord server to share more about this and other fun stuff to do in Red Eclipse: https://www.redeclipse.net/discord

**Reply by ShiroWork-s** [reply]
Ok thanks. Excited to see it in action.
I'll continue watching this with great interest ^-^

**Reply by ShiroWork-s** [reply]
Also, what about breakable glass and such when you shoot it? That seems to be a good add to any fps.

**Reply by qreeves** [reply]
@SniperGoth Yeah, how about it? 😛

**Reply by SniperGoth** [reply]
I mean, that could be done if i made a destruction simulation on Blender, then export as a MD3 model i guess.
Would need to have a new actor specifically for that to work tho.

**Reply by ShiroWork-s** [reply]
That'd be cool, but thinking of the entity mass, would you create the glass shards as separate entities, or all as one grouped entity? Cause separate entities would create realistic falling and rolling around animations, yet would cause lag spikes depending on its quality, and one grouped entity might cause it so the shards would clip through the ground or have a weird collider..
I recommend the separate entities but no shards, like a glass mist or something. I've seen games with effects of when you shoot the glass, its sprays dust and shrapnel away from the hit point, yet when hit again it removes all the shrapnel and dust and creates another one. Good for fps's that focus on gameplay and not quality of the surrounding effects.

**Reply by ShiroWork-s** [reply]
Ack, to make my notes easier to read, heres it shortened: (not really just more understandable)

Subject: Add dust and shrapnel from the bullet impact, and create shrapnel flying instead of watching shards form and fall like other games, or having the glass shards and dust instantly disappear. Extra: maybe add breakable ammo crates and auto pickup for the ammo that is shot out.
Details:
You know that to reduce lag you could make a destruction animation with one entity but fixed animation, (a picked animation that happens all the time unless randomized) but to make the game more detailed you could make separate entities as different glass shards to add more gravity and physic-like events, but would create more lag. My fix for this is, instead of making glass shards or anything in that category, just make dust. Dust and particles flying from the point of impact on the glass/destructible-item should look more realistic than a fixed animation of shards falling. I've seen many games where you shoot glass and it goes overboard with a bullet hole>dust>shards forming>shards falling>dust rising>barrier disabling. I think for a fast paced FPS like this, it should be a quick shatter with bullet hole>dust>shards falling>barrier disabling. Also, i believe for this state of the game, breakable ammo chests would be a great addition, for example... Running away from an enemy while low on ammo, you see a glass crate built into the wall, you shoot it and it knocks out some ammo for you to auto pick up on the go.
